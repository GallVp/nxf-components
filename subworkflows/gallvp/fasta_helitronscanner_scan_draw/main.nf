include { HELITRONSCANNER_SCAN as HELITRONSCANNER_SCAN_HEAD     } from '../../../modules/gallvp/helitronscanner/scan/main.nf'
include { HELITRONSCANNER_SCAN as HELITRONSCANNER_SCAN_TAIL     } from '../../../modules/gallvp/helitronscanner/scan/main.nf'
include { HELITRONSCANNER_DRAW                                  } from '../../../modules/gallvp/helitronscanner/draw/main.nf'

include { HELITRONSCANNER_SCAN as HELITRONSCANNER_SCAN_HEAD_RC  } from '../../../modules/gallvp/helitronscanner/scan/main.nf'
include { HELITRONSCANNER_SCAN as HELITRONSCANNER_SCAN_TAIL_RC  } from '../../../modules/gallvp/helitronscanner/scan/main.nf'
include { HELITRONSCANNER_DRAW as HELITRONSCANNER_DRAW_RC       } from '../../../modules/gallvp/helitronscanner/draw/main.nf'

workflow FASTA_HELITRONSCANNER_SCAN_DRAW {

    take:
    ch_fasta                        // channel: [ val(meta), fasta ]

    main:

    ch_versions = Channel.empty()

    // MODULE: HELITRONSCANNER_SCAN  as HELITRONSCANNER_SCAN_HEAD
    HELITRONSCANNER_SCAN_HEAD (
        ch_fasta,
        'head',     // command
        [],         // lcv_filepath
        0           // buffer_size
    )

    ch_helitronscanner_scan_head    = HELITRONSCANNER_SCAN_HEAD.out.scan
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_SCAN_HEAD.out.versions)

    // MODULE: HELITRONSCANNER_SCAN  as HELITRONSCANNER_SCAN_TAIL
    HELITRONSCANNER_SCAN_TAIL (
        ch_fasta,
        'tail',     // command
        [],         // lcv_filepath
        0           // buffer_size
    )

    ch_helitronscanner_scan_tail    = HELITRONSCANNER_SCAN_TAIL.out.scan
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_SCAN_TAIL.out.versions)

    // MODULE: HELITRONSCANNER_DRAW
    HELITRONSCANNER_DRAW (
        ch_fasta,
        ch_helitronscanner_scan_head,
        ch_helitronscanner_scan_tail
    )

    ch_helitronscanner_draw         = HELITRONSCANNER_DRAW.out.draw
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_DRAW.out.versions)

    // MODULE: HELITRONSCANNER_SCAN  as HELITRONSCANNER_SCAN_HEAD_RC
    HELITRONSCANNER_SCAN_HEAD_RC (
        ch_fasta,
        'head',     // command
        [],         // lcv_filepath
        0           // buffer_size
    )

    ch_helitronscanner_scan_head_rc = HELITRONSCANNER_SCAN_HEAD_RC.out.scan
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_SCAN_HEAD_RC.out.versions)

    // MODULE: HELITRONSCANNER_SCAN  as HELITRONSCANNER_SCAN_TAIL_RC
    HELITRONSCANNER_SCAN_TAIL_RC (
        ch_fasta,
        'tail',     // command
        [],         // lcv_filepath
        0           // buffer_size
    )

    ch_helitronscanner_scan_tail_rc = HELITRONSCANNER_SCAN_TAIL_RC.out.scan
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_SCAN_TAIL_RC.out.versions)

    // MODULE: HELITRONSCANNER_DRAW as HELITRONSCANNER_DRAW_RC
    HELITRONSCANNER_DRAW_RC (
        ch_fasta,
        ch_helitronscanner_scan_head_rc,
        ch_helitronscanner_scan_tail_rc
    )

    ch_helitronscanner_draw_rc      = HELITRONSCANNER_DRAW_RC.out.draw
    ch_versions                     = ch_versions.mix(HELITRONSCANNER_DRAW_RC.out.versions)

    emit:
    helitronscanner_draw            = ch_helitronscanner_draw       // channel: [ val(meta), draw ]
    helitronscanner_draw_rc         = ch_helitronscanner_draw_rc    // channel: [ val(meta), rc.draw ]
    versions                        = ch_versions                   // channel: [ versions.yml ]
}
