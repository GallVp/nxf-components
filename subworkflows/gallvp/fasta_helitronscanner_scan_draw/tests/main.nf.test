nextflow_workflow {

    name "Test Subworkflow FASTA_HELITRONSCANNER_SCAN_DRAW"
    script "../main.nf"
    workflow "FASTA_HELITRONSCANNER_SCAN_DRAW"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_gallvp"
    tag "subworkflows/fasta_helitronscanner_scan_draw"

    tag "helitronscanner/draw"
    tag "helitronscanner/scan"
    tag "gunzip"

    setup {
        run('GUNZIP') {
            script "../../../../modules/gallvp/gunzip/main.nf"

            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("actinidia_chinensis - genome_1_fasta_gz") {

        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }

    test("sarscov2 - genome") {

        when {
            workflow {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.helitronscanner_draw },       // empty
                { assert workflow.out.helitronscanner_draw_rc },    // empty
                { assert snapshot(
                    workflow.out.versions
                    ).match()
                }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }
}
