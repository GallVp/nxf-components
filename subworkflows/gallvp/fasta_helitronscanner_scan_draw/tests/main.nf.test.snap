{
    "actinidia_chinensis - genome_1_fasta_gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.draw:md5,434aaaa70c294464e6bae93b2895b13c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.rc.draw:md5,7aeed0fdf50f8cb2516b2e92f456835e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,42ab90da7ef86125b9dec685f4f08e84",
                    "versions.yml:md5,9b6a0f2f56d89451cf16702e2f358520",
                    "versions.yml:md5,a928ae7c94a23e5c62c27915dada660c",
                    "versions.yml:md5,c417497b1caea1cd51f8440df92daa40",
                    "versions.yml:md5,d786d80e914a86ce0361cfc0bc62cda9",
                    "versions.yml:md5,f3b6aa20e81f04fd731fad64f7ea3c14"
                ],
                "helitronscanner_draw": [
                    [
                        {
                            "id": "test"
                        },
                        "test.draw:md5,434aaaa70c294464e6bae93b2895b13c"
                    ]
                ],
                "helitronscanner_draw_rc": [
                    [
                        {
                            "id": "test"
                        },
                        "test.rc.draw:md5,7aeed0fdf50f8cb2516b2e92f456835e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,42ab90da7ef86125b9dec685f4f08e84",
                    "versions.yml:md5,9b6a0f2f56d89451cf16702e2f358520",
                    "versions.yml:md5,a928ae7c94a23e5c62c27915dada660c",
                    "versions.yml:md5,c417497b1caea1cd51f8440df92daa40",
                    "versions.yml:md5,d786d80e914a86ce0361cfc0bc62cda9",
                    "versions.yml:md5,f3b6aa20e81f04fd731fad64f7ea3c14"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-21T11:53:51.717511"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.draw:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.rc.draw:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,42ab90da7ef86125b9dec685f4f08e84",
                    "versions.yml:md5,9b6a0f2f56d89451cf16702e2f358520",
                    "versions.yml:md5,a928ae7c94a23e5c62c27915dada660c",
                    "versions.yml:md5,c417497b1caea1cd51f8440df92daa40",
                    "versions.yml:md5,d786d80e914a86ce0361cfc0bc62cda9",
                    "versions.yml:md5,f3b6aa20e81f04fd731fad64f7ea3c14"
                ],
                "helitronscanner_draw": [
                    [
                        {
                            "id": "test"
                        },
                        "test.draw:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "helitronscanner_draw_rc": [
                    [
                        {
                            "id": "test"
                        },
                        "test.rc.draw:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,42ab90da7ef86125b9dec685f4f08e84",
                    "versions.yml:md5,9b6a0f2f56d89451cf16702e2f358520",
                    "versions.yml:md5,a928ae7c94a23e5c62c27915dada660c",
                    "versions.yml:md5,c417497b1caea1cd51f8440df92daa40",
                    "versions.yml:md5,d786d80e914a86ce0361cfc0bc62cda9",
                    "versions.yml:md5,f3b6aa20e81f04fd731fad64f7ea3c14"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-21T11:54:17.824589"
    },
    "sarscov2 - genome": {
        "content": [
            [
                "versions.yml:md5,42ab90da7ef86125b9dec685f4f08e84",
                "versions.yml:md5,9b6a0f2f56d89451cf16702e2f358520",
                "versions.yml:md5,a928ae7c94a23e5c62c27915dada660c",
                "versions.yml:md5,c417497b1caea1cd51f8440df92daa40",
                "versions.yml:md5,d786d80e914a86ce0361cfc0bc62cda9",
                "versions.yml:md5,f3b6aa20e81f04fd731fad64f7ea3c14"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-21T12:03:43.135817"
    }
}